lib/base/process.cpp

```diff
@@ -1131,10 +1131,17 @@ bool Process::DoEvents()
 		Log(LogWarning, "Process")
 			<< "PID " << m_PID << " (" << PrettyPrintArguments(m_Arguments) << ") died mysteriously: waitpid failed";
 	} else if (WIFEXITED(status)) {
-		exitcode = WEXITSTATUS(status);
+		if (m_SentSigterm) {
+			exitcode = 128;
 
-		Log(LogNotice, "Process")
-			<< "PID " << m_PID << " (" << PrettyPrintArguments(m_Arguments) << ") terminated with exit code " << exitcode;
+			Log(LogNotice, "Process") << "PID " << m_PID << " (" << PrettyPrintArguments(m_Arguments)
+				<< ") terminated with exit code " << exitcode << " after sending SIGTERM";
```

Comment:
This log message feels wrong because it didn't terminate with the here mentioned exit code. I'm not sure if logging the hard coded exit code is of any use here.

Reply:
Indeed. I pushed another version, now there's a slight discrepancy in logging as the real exit code isn't part of the `ProcessResult`, but that's already the case for processes terminated by a signal:

```
[2021-07-27 18:01:38 +0200] notice/Process: PID 150 ('sh' '-c' 'trap 'printf TERM; exit 1' TERM; printf 'check start'; sleep inf & wait $!') terminated with exit code 1 after sending SIGTERM
[2021-07-27 18:01:38 +0200] warning/PluginCheckTask: Check command for object 'agent-a-1' (PID: 150, arguments: 'sh' '-c' 'trap 'printf TERM; exit 1' TERM; printf 'check start'; sleep inf & wait $!') terminated with exit code 128, output: check start<Timeout exceeded.>TERM
```